src/wechaty.ts

```diff
@@ -521,6 +568,51 @@ export class Wechaty extends EventEmitter implements Sayable {
     return this
   }
 
+  /**
+   * @param   {WechatyPlugin | WechatyPlugin[]} plugins      - The plugins you want to use
+   *
+   * @return  {Wechaty}                                      - this for chaining,
+   *
+   * @desc
+   * For wechaty ecosystem, allow user to define a 3rd party plugin for the current wechaty instance.
+   *
+   * @example
+   * // Report all chat message to my server.
+   *
+   * Class WechatyReportPlugin {
+   *   constructor(options) {
+   *     this.options = options;
+   *   }
+   *   install(bot) {
+   *     bot.on('message', message => http.post(this.options.url, { data: message }))
+   *   }
+   * }
+   *
+   * bot.use(new WechatyReportPlugin({ url: 'http://somewhere.to.report.your.data.com' });
+   */
+  public use (plugins: WechatyPlugin | WechatyPlugin[]) {
+    ([] as WechatyPlugin[]).concat(plugins).forEach(plugin => {
+      if (!this.plugins.includes(plugin)) {
+        this.plugins.push(plugin)
```

Comment:
The same comment as the `static use()` before: I'd like to be able to _set and forget_ the plugins... if it can be forgotten.

Reply:
It's can be forgotten. but I'm just thinking maybe the user wants to know what plugins they have installed.

once we forget it, user can never know what plugins they have installed.

Reply:
If it can be forgotten, how about leaving the "what plugins have I installed" responsibility to the end-user who is adding plugins?

Because:

1. They can save their plugins to an array for future use very easily (if needed)
1. `Wechaty.plugins` will make one more property in Wechaty class, which there have already a lot. (I always want to move them out to external instead of keeping them internal)

Reply:
ok. will remove the property